File: android/app/src/elt/java/ru/ubsmobile/BankTheme/BankTheme.java
package ru.ubsmobile.BankTheme;

import android.graphics.Color;

import com.facebook.react.bridge.ReactApplicationContext;

public class BankTheme extends BankThemeBase {

    public BankTheme(ReactApplicationContext context) {
        super(context);
    }

    public static String BackgroundColor = "#A52A2A";

    public void requestAutostartPermission() {
        super.requestAutostartPermission();
    }

    @Override
    protected String Color1() {
        return "#A52A2A";
    }

    @Override
    protected boolean AllowShowCvvCode() { return false; }

    @Override
    protected String NavigationBackgroundColor() {
        return "#A52A2A";
    }

    @Override
    protected MapLocation DefaultMapLocation() {
        return new MapLocation("54.511425", "36.259059");
    }

    @Override
    protected boolean ShowBonusesPage() {
        return false;
    }

    @Override
    protected boolean ShowNotificationsSetting() {
        return true;
    }

    @Override
    protected boolean AllowAddCardsToWallet() {
        return false;
    }

    @Override
    protected boolean ShowPersonalOffersOnMyBankPage() {
        return true;
    }

    @Override
    protected CodeSettingPageTheme CodeSettingPageTheme() {
        CodeSettingPageTheme c = new CodeSettingPageTheme();
        c.backgroundColor = "#fce3d7";
        c.buttonBorderColor = Color1();
        c.buttonTextColor = Color1();
        c.textColor = Color1();
        c.indicatorBorderColor = Color1();
        c.indicatorEmptyBackgroundColor = "transparent";
        c.indicatorFullBackgroundColor = Color1();
        return c;
    }

    @Override
    protected String BankPhoneNumber() {
        return "+74842721465";
    }
  
    @Override
    protected boolean PushNotificationsUsed() { return false; }

    @Override
    protected boolean BankMessagesUsed() { return true; }

}

File: ios/elt_app/BankTheme.swift

public class BankTheme: NSObject, BankThemeBase {
  public var color1: String = "#A52A2A";
  public var serverUrl: String = "https://dbo.bankelita.ru:444";
  public var udSuiteName: String = "elt";
  public var allowShowCvvCode = false;
  public var showNotificationsSetting = true
  public var showBonusesPage: Bool = false
  public var statusBarTheme: String = "light"
  public var navigationBackgroundColor: String = "#A52A2A"
  public var allowAddCardsToWallet = false
  public var bankPhoneNumber = "+74842721465"
  public var showPersonalOffersOnMyBankPage = true
  public var bankMessagesUsed: Bool = true
  public var pushNotificationsUsed: Bool = false
  public var defaultMapLocation: MapLocation = MapLocation(latitude: "54.511425", longitude: "36.259059")

  public var codeSettingPageTheme: CodeSettingPageTheme?

  override init() {
    self.codeSettingPageTheme = CodeSettingPageTheme(backgroundColor: "#fce3d7",
                                                     buttonBackgroundColor: self.color1,
                                       buttonBorderColor: self.color1,
                                       buttonTextColor: self.color1,
                                       textColor: self.color1,
                                       indicatorBorderColor: self.color1,
                                       indicatorEmptyBackgroundColor: "transparent",
                                       indicatorFullBackgroundColor: self.color1
    )
  }
}


